projects/wizard_of_wikipedia/interactive_end2end/interactive_end2end.py

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/env python
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Hooks up the components for the wizard generator to do live retrieval.
+"""
+
+from parlai.core.agents import Agent, create_agent, create_agent_from_shared
+from projects.wizard_of_wikipedia.generator.agents import EndToEndAgent
+
+from parlai.tasks.wizard_of_wikipedia.agents import TOKEN_KNOWLEDGE
+
+import json
+import os
+
+
+class InteractiveEnd2endAgent(Agent):
```

Comment:
can this inherit from the retrieval one or nah?

i can open an issue to move the tfidf stuff into the world

Reply:
That'd be great.

So some of the issues:

- We need a way for the agent to mark knowledge as not to be picked again. When the knowledge was handled separately by the two stage agent, that was easy enough
- The generative model expects that KNOWLEDGE token to be separating them, along with a title.